### PR TITLE
docs: add tutorial

### PR DESCRIPTION
**Pull request by Doug-AWS**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

First cut of API Gateway/Lamdba tutorial based on Romain's video.

### CODE REVIEW

File: docs/src/tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+
+.. code-block:: ts
+
+    import apple_service = require('../lib/apple_service');
+
+Add the following line of code at the end of the constructor in *my_apple_service.ts*.
+
+.. code-block:: ts
+
+    new apple_service.AppleService(this, 'Apples');
```
Comment (eladb) [inline docs/src/tutorial.rst]: Provide more context here as to where this needs to be added? I believe it needs to be added to the stack's constructor right?

File: docs/src/tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+    cdk synth
+
+.. _deploy_and_test:
+
+Step 5: Deploy and Test the App
+===============================
+
+Before you can deploy your app the first time,
+you must bootstrap your deployment,
+which creates some AWS infracture that the |cdk|
```
Comment (eladb) [inline docs/src/tutorial.rst]: Reference the docs that explain what this is

File: docs/src/tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+
+              S3.listObjectsV2({ Bucket: bucketName })
+                .promise()
+                .then(function(data) {
+                  var body = { apples: data.Contents.map(function(e) { return e.Key }) };
+                  console.log("Body for GET /: " + body);
+                  var response = makeResponse(body);
+                    console.log("GET / response: \n" + response);
+                  callback(null, response);
+              })
```
Comment (eladb) [inline docs/src/tutorial.rst]: Indentation

File: docs/src/tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+              // GET /name
+              name = getName(event.path);
+
+              S3.getObject({ Bucket: bucketName, Key: name})
+                .promise()
+                .then(function(data) {
+                  var body = data.Body.toString('utf-8');
+                  var response = makeResponse(body);
+                  callback(null, response);
+              })
```
Comment (eladb) [inline docs/src/tutorial.rst]: Indentation

File: docs/src/tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+
+      return response;
+    }
+
+Wire these functions up to our API Gateway code in *apple_service.ts*
+by adding the following code at the end of the constructor.
+
+.. code-block:: ts
+
+    const apple = api.root.addResource('{name}');
```
Comment (eladb) [inline docs/src/tutorial.rst]: Usually `POST` is done on the parent (in this case `api.root`), with any attributes passed in the application/json request body:

```
POST /apples
{
  "name": "my-new-apple"
}
```

File: docs/src/tutorial.rst [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+    import cdk = require('@aws-cdk/cdk');
+    import apigateway = require('@aws-cdk/aws-apigateway');
+    import lambda = require('@aws-cdk/aws-lambda');
+    import s3 = require('@aws-cdk/aws-s3');
+
+    export class AppleService extends cdk.Construct {
+      constructor(parent: cdk.Construct, name: string) {
+        super(parent, name);
+
+        const bucket = new s3.Bucket(this, 'AppleStore');
```
[the remark this answers is not in the document]
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: apple -> widget

I also had another writer wade through the topic.

File: docs/src/tutorial.rst [hunk 7]
```diff
@@ -0,0 +1,505 @@
...
+
+        const handler = new lambda.Function(this, 'AppleHandler', {
+          runtime: lambda.Runtime.NodeJS610,
+          code: lambda.Code.directory('resources'),
+          handler: 'apples.main',
+          environment: {
+            BUCKET: bucket.bucketName
```
[the remark this answers is not in the document]
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: done

File: docs/src/tutorial.rst [hunk 8]
```diff
@@ -0,0 +1,505 @@
...
+
+        bucket.grantReadWrite(handler.role);
+
+        const api = new apigateway.RestApi(this, 'apples-api', {
+          restApiName: 'Apple Service',
+          description: 'This service serves apples.'
+        });
+
+        const getApplesIntegration = new apigateway.LambdaIntegration(handler, {
+	  requestTemplates:  { "application/json": '{ "statusCode": "200" }' }
```
[the remark this answers is not in the document]
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: It's a left-over from Romain's code. I fixed the indentation.

[the remark this answers is not in the document]
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: I've removed the requestTemplates prop as they are not necessary.

File: docs/src/tutorial.rst (hunk 1 above)
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: The text before:

Add the following line of code at the end of the constructor in *my_widget_service.ts*.

File: docs/src/tutorial.rst (hunk 2 above)
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: I've added a link to the tools topic.

File: docs/src/tutorial.rst (hunk 3 above)
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: I've caught all tabs.

File: docs/src/tutorial.rst (hunk 4 above)
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: Here too

File: docs/src/tutorial.rst (hunk 5 above)
Comment (Doug-AWS) [reply docs/src/tutorial.rst]: You are correct, but the example is so simple, requiring just the name.

**Pull request closed without merging**